### DISCUSSION TITLE
auto_load_trailblazer_state_on_enter without automatically opening buffer?

### DISCUSSION THREAD

**Discussion by serranomorante** [LeonHeidelbach/trailblazer.nvim#51] [Q&A] [answered]
Hi! I'm starting to like this plugin a lot. I've activated `auto_load_trailblazer_state_on_enter = true` and noticed that it automatically opens the buffer with a trail mark on load. Is there any way to deactivate this behavior and still load the trail marks without trailblazer automatically opening a buffer on my behalf? I use another plugin that restores buffers for me on vim startup (`resession.nvim`) and sometimes my last opened file was a buffer without trail marks, that means resession is trying to open buffer1 (which has no trail marks) and trailblazer is also trying to open buffer10 (last buffer with trail marks)...

**Comment by serranomorante** [comment]
Ok, I managed to fix this by patching trailblazer's function `focus_win_and_buf` when loading sessions from both `resession` and `trailblazer` and then unpatching it after that.

I haven't battle-tested this but works fine for now.

> trailblazer config

```lua
return {
  "LeonHeidelbach/trailblazer.nvim",
  lazy = true,
  dependencies = "nvim-treesitter/nvim-treesitter",
  opts = {
    auto_save_trailblazer_state_on_exit = false, -- we are manually doing it on resession.nvim
    auto_load_trailblazer_state_on_enter = false,  -- we are manually doing it on resession.nvim
...
}
```

> resession config

```lua
return {
  "stevearc/resession.nvim",
  dependencies = "LeonHeidelbach/trailblazer.nvim",
  config = function(_, opts)
    local resession = require("resession")
    local trailblazer = require("trailblazer")
    local trailblazer_common = require("trailblazer.trails.common")
    local focus_win_and_buf = trailblazer_common.focus_win_and_buf

    resession.setup(opts)
    resession.add_hook("pre_load", function()
      ---Patch trailblazer
      trailblazer_common.focus_win_and_buf = function() return true end
    end)
    resession.add_hook("post_load", function()
      trailblazer.load_trailblazer_state_from_file()
      ---Unpatch trailblazer
      vim.schedule(function() trailblazer_common.focus_win_and_buf = focus_win_and_buf end)
    end)

    ...
    vim.api.nvim_create_autocmd("VimLeavePre", {
      desc = "Save a dir-specific session when you close Neovim",
      group = vim.api.nvim_create_augroup("autosave_session", { clear = true }),
      callback = function()
        -- Only save the session if nvim was started with no args
        if vim.fn.argc(-1) == 0 then
          trailblazer.save_trailblazer_state_to_file()
          resession.save_tab(vim.fn.getcwd(), { dir = "dirsession", notify = false })
        end
      end,
    })
  end,
...
```

**Comment by LeonHeidelbach** [comment, marked as answer]
@serranomorante I'm glad you like the plugin. I have considered implementing loading buffers on demand, i.e. only when jumping to a specific mark, however this would have required some rather more involved changes to the code base and did also cause several issues when I was experimenting with it. Since I would not use this myself I decided agains implementing this functionality. However, if you would like to work on this and create a PR, feel free to do so and I will review it. If the proposed changes for your specific setup is all you need, that's also fine :+1: .

**Reply by serranomorante** [reply]
thank you! yes, I prefer a different plugin (`resession.nvim`) to be in charge of restoring the buffers for me and only later trigger trailblazer's `load_trailblazer_state_from_file()` (with `focus_win_and_buf` patched to an empty function). Doing it this way I'm successfully restoring the trail marks and still preventing trailblazer from trying to open any buffer (because resession already did that). So yeah, my patch seems to be enough. Thank you so much again, this plugin is very well done 💖
